### ISSUE TITLE
QFile events for exceeding maxFileSize and maxTotalSize

### ISSUE THREAD

**Issue by FrittenKeeZ** [quasarframework/quasar#6961]
**Is your feature request related to a problem? Please describe.**
Currently, if a user chooses a file bigger than the size limitations, the selection is discarded and no information why is given to the user.

**Describe the solution you'd like**
Either implement out-of-the-box errors when exceeding `maxFileSize` or `maxTotalSize`, or provide events for manually handling the errors.

**Describe alternatives you've considered**
Using a hint to display a message about too big files being discarded.

**Additional context**
This applies to QUploader as well.

**Comment by pdanpdan** [comment]
Maybe also add "A new property append that only works with multiple that allows new files to be appended" from https://github.com/quasarframework/quasar/issues/6955

**Comment by hawkeye64** [comment]
@pdanpdan give this a go - may need further tweaking/optimization

```js
    __addFiles (e, fileList) {
      if (this.value !== null) {
        const value = this.value
        fileList = fileList !== void 0
          ? fileList
          : e.target.files
        value.concat(...fileList)
        fileList = value
      }

      const files = this.__processFiles(e, fileList)

      files !== void 0 && this.__emitValue(
        this.maxFiles !== void 0
          ? files.slice(0, parseInt(this.maxFiles, 10))
          : files
      )
    },
```

**Comment by rstoenescu** [comment]
New event ("@rejected") will be available in "quasar" v1.11.2
